util/host-utils.c

```diff
@@ -266,3 +266,154 @@ void ulshift(uint64_t *plow, uint64_t *phigh, int32_t shift, bool *overflow)
         *plow = *plow << shift;
     }
 }
+/*
+ * Unsigned 256-by-128 division.
+ * Returns the remainder via r.
+ * Returns lower 128 bit of quotient.
+ */
+static Int128 udiv256_qrnnd(Int128 *r, Int128 n0, Int128 n1, Int128 d)
+{
+    Int128 d0, d1, q0, q1, r1, r0, m;
+    uint64_t mp0, mp1;
+
+    d0 = int128_make64(int128_getlo(d));
+    d1 = int128_make64(int128_gethi(d));
+
+    r1 = int128_remu(n1, d1);
```

Comment:
When CONFIG_INT128 is not defined, int128_remu + int128_divu will end up calling int128.c's divrem128 twice, and that's an expensive operation. Not sure what would be the best approach, but you might want to consider exposing a divrem interface from int128.h (for both CONFIG_INT128 and !CONFIG_INT128) and using it here